ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -377,12 +382,17 @@ class ( ShowLedgerState (LedgerTables l)
   -- 'CardanoBlock' instance uses 'projectLedgerTables'.
   withLedgerTables :: IsApplyMapKind mk => l any -> LedgerTables l mk -> l mk
 
+  projectMk :: LedgerTables l mk -> mk (TableKey l) (TableValue l)
+
+  injectMK  :: mk (TableKey l) (TableValue l) -> LedgerTables l mk
+
   pureLedgerTables ::
        (forall k v.
             (Ord k, Eq v)
          => mk k v
        )
     -> LedgerTables l mk
+  pureLedgerTables = injectMK
```

Comment:
I wonder if we still need to quantify over all `k` and `v` if we now have `TableKey l` and `TableValue l`. For example:
```suggestion
  pureLedgerTables ::
       mk (TableKey l) (TableValue l)
    -> LedgerTables l mk
  pureLedgerTables = injectMK
```

Trying this out locally, I found that this does cause problems for the definition of `pureLedgerTables` for `Ledger.Dual`

Reply:
Sorry I'm no longer using `TableValue` and `TableKey`